docs/sdkx-react-native/delegates.mdx

```diff
@@ -0,0 +1,227 @@
+---
+sidebar_position: 100
+title: Helpshift Delegates
+description: "The Helpshift SDK provides delegate callbacks to help app developers track a user's activities within the help section."
+---
+
+import {
+  Admonition,
+  CodeBlock,
+  Tabs,
+  TabItem,
+  LatestSdkVersion,
+  Centered,
+  Image,
+  Intro,
+  SideBySide,
+  DownloadButton,
+  Steps,
+  Step,
+} from "@site/src/components/forDocs";
+
+# Helpshift Delegates {#delegates}
+
+<Intro>
+
+The Helpshift SDK provides delegate callbacks to help app developers track a user's activities within the help section.
+
+</Intro>
+
+## Helpshift Delegate implementation {#delegate-implementation}
+
+- To use Helpshift Delegate please import `helpshiftEmitter`
+```
+import {
+  helpshiftEmitter,
+} from 'helpshift-plugin-sdkx-react-native';
+
+```
+
+```
+const handleCallbacks = () => {
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'receivedUnreadMessageCount') {
+        try {
+          let countData =
+            Platform.OS == 'ios' ? data.eventData : JSON.parse(data.eventData);
+          console.log(countData.count);
+        } catch (err) {
+          console.error('Parsing Error', err);
+        }
+      } else {
+        console.log(JSON.stringify(data));
+      }
+    });
+    helpshiftEmitter.addListener('onUserAuthenticationFailure', (data) => {
+      Toast.show({
+        type: 'error',
+        text1: JSON.stringify(data),
+      });
+    });
+  };
+```
+
+
+## Events {#helpshift-events}
+
+### Helpshift session started event {#session-delegates}
+
+This event gets fired when the Helpshift session starts
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'helpshiftSessionStarted') {
+        //Your code here.
+      }
+    });  
+```
+
+### Helpshift session ended event {#session-delegates}
+
+This event gets fired when the Helpshift session ends.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'helpshiftSessionEnded') {
+        //Your code here.
+      }
+    });  
+```
+
+### Unread Message Count Event {#unreadMessageCount}
+
+This event gets fired when you call `RequestUnreadMessageCount(FetchFromServer)` API to request unread messages count.
+
+```
+    helpshiftEmitter.addListener('onEventOccurred', (data) => {
+      if (data.eventName == 'receivedUnreadMessageCount') {
+        //Your code here.
+      } 
+    });
+    helpshiftEmitter.addListener('onUserAuthenticationFailure', (data) => {
+       //Your code here.
+    });
+```
+### Conversation Status Event {#conversation-status}
+
+This event contains information about the current ongoing conversation.
```

Comment:
When is this event fired ?
Please mention that.

Reply:
Hi @sumeet-helpshift  there is no info provided regarding any starting point in any doc in android or ios.

Reply:
@vikashelpshift : Please find out when this event is fired from webchat people